src/solution/server/services/todo_manager.js

```diff
@@ -0,0 +1,85 @@
+const fs = require("fs").promises;
+const todoFile = "./data/todo_list.json";
+
+async function getAll() {
+  const data = await readTodoFile();
+  return data;
+}
+
+async function addTodo(text) {
+  const data = await readTodoFile();
+  let id = 1;
+  if (data.length) id = data[data.length - 1].id + 1;
+  const newTodo = {
+    id,
+    text,
+    time: new Date().toLocaleDateString(),
+    complete: false,
+    checkTime: null,
+  };
+  if (!data) {
+    data = [];
+  }
+  data.push(newTodo);
+  await writeTodoFile(data);
+  return data;
+}
+
+async function updateTodo(todoId, fields) {
+  const data = await readTodoFile();
+  const newData = data.map((item) =>
+    item.id === todoId
+      ? {
+          ...item,
+          ...fields,
+        }
+      : item
+  );
+  await writeTodoFile(newData);
+  return newData;
```

Comment:
this function should return ONLY the updated todo task, not the whole list of todos.